common/tests/gballoc_ll_mimalloc_ut/gballoc_ll_mimalloc_ut.c

```diff
@@ -0,0 +1,186 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#ifdef __cplusplus
+#include <cstdlib>
+#include <cstddef>
+#else
+#include <stdlib.h>
+#include <stddef.h>
+#endif
+
+#include "azure_macro_utils/macro_utils.h"
+#include "testrunnerswitcher.h"
+
+static TEST_MUTEX_HANDLE g_testByTest;
+
+static void* TEST_MALLOC_RESULT = (void*)0x1;
+static void* TEST_CALLOC_RESULT = (void*)0x2;
+static void* TEST_REALLOC_RESULT = (void*)0x3;
+
+#include "umock_c/umock_c.h"
+
+#define ENABLE_MOCKS
+#include "umock_c/umock_c_prod.h"
+#ifdef __cplusplus
+extern "C" {
+#endif
+    MOCKABLE_FUNCTION(, void*, mock_mi_malloc, size_t, size);
+    MOCKABLE_FUNCTION(, void*, mock_mi_calloc, size_t, nmemb, size_t, size);
+    MOCKABLE_FUNCTION(, void*, mock_mi_realloc, void*, ptr, size_t, size);
+    MOCKABLE_FUNCTION(, void, mock_mi_free, void*, ptr);
+#ifdef __cplusplus
+}
+#endif
+#undef ENABLE_MOCKS
+
+MU_DEFINE_ENUM_STRINGS(UMOCK_C_ERROR_CODE, UMOCK_C_ERROR_CODE_VALUES)
+
+#include "azure_c_pal/gballoc_ll.h"
+
+static void on_umock_c_error(UMOCK_C_ERROR_CODE error_code)
+{
+    ASSERT_FAIL("umock_c reported error :%" PRI_MU_ENUM "", MU_ENUM_VALUE(UMOCK_C_ERROR_CODE, error_code));
+}
+
+BEGIN_TEST_SUITE(gballoc_ll_mimalloc_ut)
+
+TEST_SUITE_INITIALIZE(TestClassInitialize)
+{
+    g_testByTest = TEST_MUTEX_CREATE();
+    ASSERT_IS_NOT_NULL(g_testByTest);
+
+    umock_c_init(on_umock_c_error);
+
+    REGISTER_GLOBAL_MOCK_RETURN(mock_mi_malloc, TEST_MALLOC_RESULT);
+    REGISTER_GLOBAL_MOCK_RETURN(mock_mi_calloc, TEST_CALLOC_RESULT);
+    REGISTER_GLOBAL_MOCK_RETURN(mock_mi_realloc, TEST_REALLOC_RESULT);
+}
+
+TEST_SUITE_CLEANUP(TestClassCleanup)
+{
+    umock_c_deinit();
+
+    TEST_MUTEX_DESTROY(g_testByTest);
+}
+
+TEST_FUNCTION_INITIALIZE(TestMethodInitialize)
+{
+    if (TEST_MUTEX_ACQUIRE(g_testByTest))
+    {
+        ASSERT_FAIL("our mutex is ABANDONED. Failure in test framework");
+    }
+
+    umock_c_reset_all_calls();
+}
+
+TEST_FUNCTION_CLEANUP(TestMethodCleanup)
+{
+    TEST_MUTEX_RELEASE(g_testByTest);
+}
+
+/*Tests_SRS_GBALLOC_LL_MIMALLOC_02_001: [ gballoc_ll_init shall return 0. ]*/
+TEST_FUNCTION(gballoc_ll_init_returns_0)
+{
+    ///arrange
+    int result;
+
+    ///act
+    result = gballoc_ll_init(NULL);

```

Comment:
and what happens if the parameters are non-NULL?

Reply:
As explained in the .md "`gballoc_ll_init` returns. `params` exists as a placeholder and is ignored."

---
In reply to: [457702426](https://github.com/Azure/azure-c-pal/pull/11#discussion_r457702426) [](ancestors = 457702426)

Reply:
Can we have a test with non-NULL?

Reply:
Yes, we can. Here and everywhere added.

---
In reply to: [457763832](https://github.com/Azure/azure-c-pal/pull/11#discussion_r457763832) [](ancestors = 457763832)